pkg/operator/ceph/file/dependent.go

```diff
@@ -55,8 +58,18 @@ func cephFilesystemDependents(clusterdCtx *clusterd.Context, clusterInfo *client
 	nsName := fmt.Sprintf("%s/%s", filesystem.Namespace, filesystem.Name)
 	baseErrMsg := fmt.Sprintf("failed to get dependents of CephFilesystem %q", nsName)
 
+	deps := dependents.NewDependentList()
+	_, err := cephclient.GetFilesystem(clusterdCtx, clusterInfo, filesystem.Name)
+	if err != nil {
+		if code, ok := kexec.ExitStatus(err); ok && code == int(syscall.ENOENT) {
+			logger.Infof("the deletion will continue without checking for dependencies since the filesystem CR was not found %q", nsName)
+			return deps, nil
+		}
+		return deps, errors.Wrapf(err, "failed to check for existence of CephFilesystem %q", nsName)
+	}
```

Comment:
CephFilesystemSubvolumeGroup are tracked at the k8s level and not the ceph level and are still valid dependencies even if Ceph doesn't have a record of the underlying fs. Even if the filesystem doesn't exist in Ceph, we still need to check if CephFilesystemSubvolumeGroup CRs exist that reference this CephFilesystem so that we enforce that users need to delete those before the referenced CephFileystem can be deleted.

I'd suggest making a `filesystemExists()` helper and using that in a conditional within this function to improve readability.

```go
func filesystemExists(/*...*/) (bool, error) {
	_, err := cephclient.GetFilesystem(clusterdCtx, clusterInfo, filesystem.Name)
	if err != nil {
		if code, ok := kexec.ExitStatus(err); ok && code == int(syscall.ENOENT) {
			logger.Infof("the deletion will continue without checking for dependencies since the filesystem CR was not found %q", nsName)
			return false, nil
		}
		return false, errors.Wrapf(err, "failed to check for existence of CephFilesystem %q", nsName)
	}
	return true, nil
}
```

```go
	fsExists, err := filesystemExists(/*...*/)
	if err != nil {
		deps, err
	}

	// subvolume groups that contain subvolumes
	if fsExists {
		deps, err = subvolumeGroupDependents(clusterdCtx, clusterInfo, filesystem)
		if err != nil {
			return deps, errors.Wrapf(err, baseErrMsg)
		}
	}
```

Reply:
Thanks for your suggestion,  if only improve for readability, no needed,

Reply:
Please re-read my comment. It is not just for readability. We should still check CephFilesystemSubvolumeGroup CRs even if the underlying filesystem doesn't exist.

Reply:
@BlaineEXE done, please take a look